client/my-sites/patterns/lib/get-pattern-permalink.ts

```diff
@@ -19,3 +19,14 @@ export function getPatternPermalink(
 	url.hash = `#pattern-${ pattern.ID }`;
 	return url.toString();
 }
+
+export const extractPatternIdFromHash = () => {
+	const pattern = /^#pattern-(\d+)$/;
+	const match = window.location.hash.match( pattern );
+
+	if ( match ) {
+		return parseInt( match[ 1 ], 10 );
+	}
+
+	return undefined;
+};
```

Comment:
I'm on the fence - I can understand why we would put that here given it relates to pattern permalinks. However, the file is named after the other function so it's a bit confusing to see it also exports that new function. Said differently: should we rename this file to something more generic, or move that new function to it's own file?

I think we should also revisit whether we should keep `paths.ts` if some functions that relate to paths/urls are in a dedicated file while some others are located in that particular file.